server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/RemarksNode.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.model.tree;
+
+import lombok.Getter;
+import lombok.ToString;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.tree.statements.ObsoleteNode;
+
+/** The class represents the Remarks Node. */
+@ToString(callSuper = true)
+@Getter
+public class RemarksNode extends ObsoleteNode {
+  NodeType nodeType;
+
+  public RemarksNode(Locality locality, NodeType nodeType) {
+    super(locality);
+    this.nodeType = nodeType;
+    addProcessStep(this::throwObsoleteNodeWarning);
```

Comment:
You can move this call to the constructor of the super class. In fact, this class should not have any specific logic, it will be used only for the class-based analysis